web-server/src/components/AnimatedInputWrapper/AnimatedInputWrapper.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { themeColors } from "@/theme/schemes/theme";
+
+const AnimatedInputWrapper = () => {
+  const [isAnimationVisible, setIsAnimationVisible] = useState(true);
+  const [value, setValue] = useState("");
+
+  const addAnimation = useCallback(
+    () => setIsAnimationVisible(true),
+    [setIsAnimationVisible]
+  );
+
+  const handleChange = useCallback(
+    (e: any) => setValue(e.target.value),
+    [setValue]
+  );
+
+  return (
+    <div>
+        {isAnimationVisible && !value && <AnimatedPlaceHolderWrapper />}
+        <input
+          value={value}
+          onChange={handleChange}
+          onBlur={addAnimation}
+          placeholder={!isAnimationVisible ? "Search" : ""}
+          style={{backgroundColor: themeColors.primaryAlt,border: "none",outline: "none", right: "0px"}}
+        />
+    </div>
+    
+  );
+};
+
+const popRepos = ["golang/go", "mozilla/rust", "apple/swift", "oven-sh/bun"];
+
+const AnimatedPlaceHolderWrapper = () => {
+  return (
+    <div
+      style={{
+        position: "absolute",
+        top: "50%",
+        transform: "translate(0%,-50%)",
+        fontSize: "1em",
+        display: "flex",
+        gap: "4px",
+      }}
+    >
+      <span>Search for</span>
+      <AnimatedRepos />
+    </div>
+  );
+};
+
+const AnimatedRepos = () => {
+  const [animationIndex, setAnimationIndex] = useState(0);
+  const [repos, setRepos] = useState(popRepos);
+
+  const increment = () =>
+    setAnimationIndex((prev) => {
+      if (prev >= popRepos.length - 1) return 0;
+      return prev + 1;
+    });
+
+  const rearrangeArray = () => {
+    const beforeIndex = repos.slice(0, 1);
+    const fromIndexOnwards = repos.slice(1);
+    const arr = fromIndexOnwards.concat(beforeIndex);
+    setRepos(arr);
+  };
+  
+  useEffect(() => {
+  const animationInterval = setInterval(increment, 2000);
+  let repoInterval: NodeJS.Timeout | undefined;
+  const timeout = setTimeout(() => {
+    repoInterval = setInterval(rearrangeArray, 2000);
```

Comment:
Is there a reason why these functions are being from within separate setInterval functions?

I understand you're trying to stagger them by 500ms after initial load, but given my other comment where it seems you might not have to use these two approaches at all, this might get refactored too.

Reply:
The new increment function solves this.